File: backend/app/schemas/approvals.py
from datetime import date, datetime

from pydantic import BaseModel, Field


class ApprovalTaskSummaryOut(BaseModel):
    task_id: int
    claim_id: int
    claim_title: str
    employee_name: str
    category_name: str
    original_currency: str
    original_amount: float
    submitted_at: datetime | None
    sequence_order: int
    status: str
    is_actionable: bool


class ApprovalActionLogOut(BaseModel):
    id: int
    action_type: str
    actor_name: str | None
    description: str | None
    created_at: datetime


class ApprovalTaskClaimDetailOut(BaseModel):
    task_id: int
    claim_id: int
    employee_name: str
    claim_title: str
    claim_description: str | None
    category_name: str
    expense_date: date
    original_currency: str
    original_amount: float
    status: str
    current_approval_step: int | None
    logs: list[ApprovalActionLogOut]


class ApprovalTaskListResponse(BaseModel):
    tasks: list[ApprovalTaskSummaryOut]


class ApprovalDecisionRequest(BaseModel):
    comment: str | None = Field(default=None, max_length=2000)


class ApprovalDecisionResponse(BaseModel):
    task_id: int
    task_status: str
    claim_id: int
    claim_status: str

File: frontend/src/types/approvals.ts
export type ApprovalTaskStatus = "PENDING" | "APPROVED" | "REJECTED" | "SKIPPED";

export interface ApprovalTaskSummary {
  task_id: number;
  claim_id: number;
  claim_title: string;
  employee_name: string;
  category_name: string;
  original_currency: string;
  original_amount: number;
  submitted_at: string | null;
  sequence_order: number;
  status: ApprovalTaskStatus;
  is_actionable: boolean;
}

export interface ApprovalTaskListResponse {
  tasks: ApprovalTaskSummary[];
}

export interface ApprovalActionLog {
  id: number;
  action_type: string;
  actor_name: string | null;
  description: string | null;
  created_at: string;
}

export interface ApprovalReceiptContext {
  receipt_id: number;
  original_filename: string;
  file_mime_type: string;
  file_size_bytes: number;
  uploaded_at: string;
}

export interface ApprovalOcrContext {
  extraction_id: number;
  engine: string | null;
  confidence: number | null;
  parsed_fields: Record<string, unknown> | null;
  created_at: string;
}

export interface ApprovalTaskClaimDetail {
  task_id: number;
  claim_id: number;
  employee_name: string;
  claim_title: string;
  claim_description: string | null;
  category_name: string;
  expense_date: string;
  original_currency: string;
  original_amount: number;
  base_currency: string;
  converted_amount: number | null;
  exchange_rate: number | null;
  exchange_rate_provider: string | null;
  status: string;
  current_approval_step: number | null;
  pending_approver_names: string[];
  receipt: ApprovalReceiptContext | null;
  ocr_extraction: ApprovalOcrContext | null;
  logs: ApprovalActionLog[];
}

export interface ApprovalDecisionPayload {
  comment?: string | null;
}

export interface ApprovalDecisionResponse {
  task_id: number;
  task_status: ApprovalTaskStatus;
  claim_id: number;
  claim_status: string;
}
